src/libsyncengine/update_detection/update_detector/updatetreeworker.cpp

```diff
@@ -804,10 +804,9 @@ ExitCode UpdateTreeWorker::step8CompleteUpdateTree() {
 
             SyncTime lastModified =
                     _side == ReplicaSide::Local ? dbNode.lastModifiedLocal().value() : dbNode.lastModifiedRemote().value();
-            SyncName name = dbNode.nameRemote();
-            std::shared_ptr<Node> newNode =
-                    std::shared_ptr<Node>(new Node(dbNode.nodeId(), _side, name, dbNode.type(), {}, newNodeId, dbNode.created(),
-                                                   lastModified, dbNode.size(), parentNode));
+            SyncName name = _side == ReplicaSide::Local ? dbNode.nameLocal() : dbNode.nameRemote();
```

Comment:
I am tempted to remove in the same PR the block containing the line `node->setName(dbNode.nameRemote());`.

Reply:
Yes agreed: 668eb485512b429c99b19b084c400c7c97c5c4ed